--- llvm-project-14.0.0.src/clang/lib/Driver/ToolChains/Arch/AArch64.cpp.orig	2022-10-01 17:46:26.136683148 +0000
+++ llvm-project-14.0.0.src/clang/lib/Driver/ToolChains/Arch/AArch64.cpp	2022-10-01 17:47:27.950284933 +0000
@@ -273,7 +273,8 @@
       Diag << A->getAsString(Args);
   }
 
-  if (Args.getLastArg(options::OPT_mgeneral_regs_only)) {
+  // if (Args.getLastArg(options::OPT_mgeneral_regs_only)) {
+  if (true) {
     Features.push_back("-fp-armv8");
     Features.push_back("-crypto");
     Features.push_back("-neon");
